pkg/vmprovider/providers/vsphere/session/session_vm_update.go

```diff
@@ -669,11 +663,10 @@ func (s *Session) prepareVMForPowerOn(
 func (s *Session) poweredOnVMReconfigure(
 	vmCtx context.VirtualMachineContext,
 	resVM *res.VirtualMachine,
-	config *vimTypes.VirtualMachineConfigInfo,
-	classConfigSpec *vimTypes.VirtualMachineConfigSpec) error {
+	config *vimTypes.VirtualMachineConfigInfo) error {
 
 	configSpec := &vimTypes.VirtualMachineConfigSpec{}
-	UpdateConfigSpecChangeBlockTracking(config, configSpec, classConfigSpec, vmCtx.VM.Spec)
+	UpdateConfigSpecChangeBlockTracking(config, configSpec, nil, vmCtx.VM.Spec)
```

Comment:
Instead of passing the ClassSpec, we can just have this `UpdateConfigSpecChangeBlockTracking` take the desired CBT state (true/false) and update it.  That way, we don't have to split brain this method (passing non-nil and nil class spec).  Can be a followup.

Reply:
For now there are two different places can set CBT, VM spec and Class spec. If we want to get the desired CBT state, we have to get from both VM spec and class spec, which leave a lot of work to the caller :( Even if the classConfigSpec is `nil`, we can't simply saying that the desired CBT state is false and still have to compare with the current VM config. This introduces code duplication if the check is done from caller.

Reply:
Yes, the prior `UpdateConfigSpecFooBar()` pattern wasn't super well suited to adding yet another param for the class ConfigSpec; resolving the desired state as a `*T` at a higher level is something we'll likely end up doing, and may make the v1a2 work easier.